connectors/sources/confluence.py

```diff
@@ -941,50 +941,62 @@ async def _attachment_coro(self, document, access_control):
             document (dict): Formatted document of page/blogpost
             access_control (list): List of identities which have access to document
         """
-        async for attachment, download_link in self.fetch_attachments(
-            content_id=document["_id"],
-            parent_name=document["title"],
-            parent_space=document["space"],
-            parent_type=document["type"],
-        ):
-            attachment = self._decorate_with_access_control(
-                document=attachment, access_control=access_control
-            )
-            await self.queue.put(
-                (  # pyright: ignore
-                    attachment,
-                    partial(
-                        self.download_attachment,
-                        download_link[1:],
-                        copy(attachment),
-                    ),
+        try:
+            async for attachment, download_link in self.fetch_attachments(
+                content_id=document["_id"],
+                parent_name=document["title"],
+                parent_space=document["space"],
+                parent_type=document["type"],
+            ):
+                attachment = self._decorate_with_access_control(
+                    document=attachment, access_control=access_control
                 )
+                await self.queue.put(
+                    (  # pyright: ignore
+                        attachment,
+                        partial(
+                            self.download_attachment,
+                            download_link[1:],
+                            copy(attachment),
+                        ),
+                    )
+                )
+        except Exception as exception:
+            self._logger.exception(
+                f"Error while fetching attachments of {document.get('title')}: {exception}"
```

Comment:
can we add the ID, space, and type the the log? That will give us the info we need to be able to reproduce the error after the fact.